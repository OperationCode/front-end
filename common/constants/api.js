import { get, post, patch } from 'common/utils/api-utils';
import { formatUserData } from 'common/utils/formatters';

/* GET REQUESTS */
export const getUserPromise = ({ token }) => get('auth/user/', { token });
export const getCodeSchoolsPromise = () => get('api/v1/codeschools/');
export const getTeamMembersPromise = () => get('api/v1/teamMembers/');

/* POST REQUESTS */
export const createUser = ({ email, password, firstName, lastName, zipcode }) =>
  post('auth/registration/', {
    email,
    firstName,
    lastName,
    password,
    zipcode,
  }).then(({ data }) => data);

export const loginUser = ({ email, password }) =>
  post('auth/login/', {
    email,
    password,
  }).then(({ data }) => data);

export const loginSocial = (provider, body) =>
  post(`auth/social/${provider}/`, body).then(({ data }) => data);

export const passwordReset = ({ email }) =>
  post('auth/password/reset/', { email }).then(({ data }) => data);

export const passwordResetSubmit = values =>
  post('auth/password/reset/confirm/', values).then(({ data }) => data);

export const changePassword = values =>
  post('auth/password/change/', values).then(({ data }) => data);

export const confirmEmail = key => post('auth/verify-email/', key).then(({ data }) => data);

/* PATCH REQUESTS */
export const updateUser = userInfo => {
  return patch('auth/profile/', {
    ...formatUserData(userInfo),
  }).then(({ data }) => data);
};
