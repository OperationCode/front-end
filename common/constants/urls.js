export const s3 = 'https://operation-code-assets.s3.us-east-2.amazonaws.com/';
export const donateLink = 'https://secure.lglforms.com/form_engine/s/BRtP7QUKyHOyEYsZROsRew';
export const leadershipCircleLink =
  'https://secure.lglforms.com/form_engine/s/L428AQ2rrsFJQyy5Fbglvg';
