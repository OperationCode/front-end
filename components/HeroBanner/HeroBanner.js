import React from 'react';
import { string, node, bool } from 'prop-types';
import classNames from 'classnames';
import Container from 'components/Container/Container';
import styles from './HeroBanner.module.css';

HeroBanner.propTypes = {
  backgroundImageSource: string,
  className: string,
  children: node,
  isFullViewportHeight: bool,
  title: string.isRequired,
};

HeroBanner.defaultProps = {
  backgroundImageSource: '',
  className: undefined,
  children: undefined,
  isFullViewportHeight: false,
};

function HeroBanner({ backgroundImageSource, children, className, isFullViewportHeight, title }) {
  return (
    <Container
      backgroundImageSource={backgroundImageSource}
      className={classNames(styles.HeroBanner, className, {
        [styles.smallHero]: !children && !backgroundImageSource,
      })}
      isFullViewportHeight={isFullViewportHeight}
    >
      <h1 className={classNames({ [styles.underline]: children })}>{title}</h1>
      {children}
    </Container>
  );
}

export default HeroBanner;
