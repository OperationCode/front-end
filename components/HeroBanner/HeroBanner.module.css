.HeroBanner {
  padding-top: 5rem;
  min-height: 60vh;
  text-shadow: #111111 0 0 15px;
}

.HeroBanner.smallHero {
  min-height: 35vh;
}

.underline {
  border-bottom: 4px solid var(--primary);
  margin-bottom: 1rem;
  text-align: center;
}
