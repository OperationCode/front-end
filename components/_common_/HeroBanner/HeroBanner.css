.HeroBanner {
  align-items: center;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: #ffffff;
  display: flex;
  flex-direction: column;
  height: 480px;
  justify-content: center;
  text-shadow: #111111 0 0 15px;
  width: 100%;
}

.content {
  padding: 3rem;
}

/* TODO: Create <Hr /> component and use that to match drop-shadow on header */
.underline {
  border-bottom: 4px solid #ffffff;
  margin-bottom: 1rem;
  text-align: center;
}

@media only screen and (max-width: 768px) {
  .HeroBanner {
    height: 320px;
  }
}
