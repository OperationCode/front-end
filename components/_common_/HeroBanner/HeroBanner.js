import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './HeroBanner.css';

HeroBanner.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  imageSource: PropTypes.node.isRequired,
  title: PropTypes.node.isRequired,
};

HeroBanner.defaultProps = {
  children: undefined,
  className: '',
};

function HeroBanner({ children, className, imageSource, title }) {
  const dynamicBackgroundImage = { backgroundImage: `url(${imageSource})` };

  return (
    <div style={dynamicBackgroundImage} className={classNames(className, styles.HeroBanner)}>
      <div className={styles.content}>
        <h1 className={classNames({ [styles.underline]: children })}>{title}</h1>
        {children}
      </div>
    </div>
  );
}

export default HeroBanner;
