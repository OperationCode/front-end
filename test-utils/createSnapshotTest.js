/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
