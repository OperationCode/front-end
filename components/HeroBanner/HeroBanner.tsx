import { cx } from 'common/utils/cva';
import Container from 'components/Container/Container';
import { HERO_BANNER_H1 } from 'common/constants/testIDs';

export interface HeroBannerPropsType {
  /**
   * Renders a title for the banner.
   */
  title: string;
  /**
   * Sets the path for an optional background image.
   */
  backgroundImageSource?: string;
  /**
   * Applies classnames to the base `figure` element for styling.
   */
  className?: string;
  /**
   * Content to be rendered in the Container.
   */
  children?: React.ReactNode;
}

function HeroBanner({ backgroundImageSource, children, className, title }: HeroBannerPropsType) {
  return (
    <Container
      backgroundImageSource={backgroundImageSource}
      className={cx(className, 'pt-20 text-shadow-[0_0_15px_#111111] min-h-dvh')}
    >
      <h1
        className={cx({ 'border-b-4 border-b-primary text-center mb-4': children })}
        data-testid={HERO_BANNER_H1}
      >
        {title}
      </h1>
      {children}
    </Container>
  );
}

export default HeroBanner;
