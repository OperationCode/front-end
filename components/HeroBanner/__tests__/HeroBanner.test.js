import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import { s3 } from 'common/constants/urls';
import HeroBanner from '../HeroBanner';

describe('HeroBanner', () => {
  it('should render with required props', () => {
    createSnapshotTest(<HeroBanner title="Test" />);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <HeroBanner
        title="Test"
        backgroundImageSource={`${s3}heroBanner/stock_family-2.jpg`}
        className="test"
        isFullViewportHeight
      >
        Testing 123
      </HeroBanner>,
    );
  });
});
