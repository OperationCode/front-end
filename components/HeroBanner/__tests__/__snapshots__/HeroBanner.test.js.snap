// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBanner should render with many props assigned 1`] = `
<div
  className="HeroBanner test Container secondary fullViewportHeight"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(33, 48, 69, 0.65),rgba(33, 48, 69, 0.65)), url(https://operation-code-assets.s3.us-east-2.amazonaws.com/heroBanner/stock_family-2.jpg)",
    }
  }
>
  <div
    className="content"
  >
    <h1
      className="underline"
    >
      Test
    </h1>
    Testing 123
  </div>
</div>
`;

exports[`HeroBanner should render with required props 1`] = `
<div
  className="HeroBanner smallHero Container secondary"
>
  <div
    className="content"
  >
    <h1
      className=""
    >
      Test
    </h1>
  </div>
</div>
`;
